### ISSUE TITLE
Should make sure api token and user/pass creds are for the same account

### ISSUE THREAD

**Issue by pR0Ps** [pR0Ps/stravaweblib#2]

**Comment by dlenski** [comment]
After logging in via web interface, the client gets 3 cookies from Strava:

```py
import stravaweblib, base64

client = stravaweblib.WebClient(access_token, email=email, password=password)
print('\n'.join("{}={} (until {})".format(ck.name, ck.value, ck.expires and datetime.datetime.fromtimestamp(ck.expires))
      for ck in client._session.cookies))

# ->
# _strava4_session=[base32 string] (until None)
# strava_remember_id=[integer subscriber ID] (until 2021-10-31 08:52:29)
# strava_remember_token=[JSON web token] (until 2021-10-31 08:52:29)

print('\n'.join(base64.b64decode(chunk, validate=False).decode() 
      for chunk in client._session.cookies['strava_remember_token'].split('.')[:2]))

# ->
# {"typ":"JWT","alg":"HS256","signing_key":"v1","encryption_key":"v1","iv":bytes_base64_enc}
# {"iss":"com.strava.athletes","sub":integer_subscribed_id,"iat":login_timestamp_seconds,
#  "exp":expiration_timestamp_seconds,"email":some_sort_of_encrypted_string}
```

As far as I can tell:
1. The `_strava4_session` cookie is redundant. If I delete it, it'll be recreated on the next request.
2. The `strava_remember_id` cookie is redundant but _must exist_. If I delete it, I'll get redirected to login, but I can recreate it from the sub(scriber) field of the [JWT](https://jwt.io) cookie.

So, the way to do this task is pretty straightforward, I think. After login, verify that the `strava_remember_id` cookie and/or the sub(scriber) field of the `strava_remember_token` cookie are equal to `client.get_athlete().id`.

**Comment by pR0Ps** [comment]
Thanks, this it a lot cleaner and faster than what I had hacked up for this issue. I was doing a `HEAD` request to `https://www.strava.com/me` and pulling the athlete id from the `Location` header that it returned.

**Comment by dlenski** [comment]
One issue with this is that the base `stravalib` client can be used in a "not-logged-in" state, where you _don't_ yet have an `access_token`, but are going to generate one with `client.exchange_code_for_token(client_id=1234, client_secret='asdf1234', code=code)`.

Perhaps `stravaweblib` should *only* do this check in the case where the `access_token` has been provided?

**Comment by pR0Ps** [comment]
Closed by #7
